ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/storage/DiffBasedWorldStateKeyValueStorage.java

```diff
@@ -122,6 +136,12 @@ public Optional<Hash> getWorldStateBlockHash() {
         .map(Hash::wrap);
   }
 
+  public Optional<Long> getWorldStateBlockNumber() {
+    return composedWorldStateStorage
+        .get(TRIE_BRANCH_STORAGE, WORLD_BLOCK_NUMBER_KEY)
+        .map(bytes -> Long.decode("0x" + new String(bytes, StandardCharsets.UTF_8)));
```

Comment:
I think you can use something like that to decode and encode in the database

encode to byte[] before writing into the database
 >  byte[] bytes = Bytes.ofUnsignedLong(100000).toArrayUnsafe();

decode from the database
> Bytes.wrap(bytes).toLong();